driver-core/src/test/unit/com/mongodb/internal/connection/ScramShaAuthenticatorSpecification.groovy

```diff
@@ -387,6 +387,24 @@ class ScramShaAuthenticatorSpecification extends Specification {
         async << [true, false]
     }
 
+    def 'should skip empty exchange on server authentication SHA-256'() {
+        given:
+        def serverResponses = createMessages('''
+            S: r=rOprNGfwEbeRWgbNEkqO%hvYDpWUa2RaTCAfuxFIlj)hNlF$k0,s=W22ZaJ0SNY7soEsUEjb6gQ==,i=4096
+            S: v=6rriTRBi23WpRR/wtup+mMhUZUn/dB5nLTJRsjl95G4=
+        ''').last()
+        def authenticator = new ScramShaAuthenticator(SHA256_CREDENTIAL, { 'rOprNGfwEbeRWgbNEkqO' }, { 'pencil' })
+
+        when:
+        authenticate(createConnection(serverResponses, 1), authenticator, async)
```

Comment:
This test passes even if you roll back the change to SaslAuthenticator.  It's because the test is just calling `authenticate` when it should be calling `validateAuthentication`

This is why you should generally write the test first to ensure it fails before making the change to the code under test, then make the code change and ensure that the test passes.   TDD FTW!

Reply:
Tests have been updated.